### ISSUE TITLE
Xcode 10b2: (intermittent) Never call this for file that sourcekitd fails.: file File+Cache.swift, line 127 Abort trap: 6

### ISSUE THREAD

**Issue by billThePill** [realm/SwiftLint#2273]
### Bug Report

I am having an intermittent failure of swiftlint when using Xcode 10 beta 2. 

Version: 0.26.0
Installation: package as well as from the source

This occurs as part of a Run Script Phase as part of the Xcode build but I can reproduce it from the command line after setting $TOOLCHAIN_DIR. It occurs with --no-cache option as well. When it does occur, it happens before linting begins on any of the files and seems to come from the extension to [SwiftVersion](https://github.com/realm/SwiftLint/blob/a35f1a9d0954cadadcccce3b65c7ba00cec8c4f6/Source/SwiftLintFramework/Models/SwiftVersion.swift) (see below). I just thought I'd pass this along to powers that be. I could not find as a similar report and it may be because of the beta of SourceKit. It there is something I can do to avoid this, please let me know.

```
$ swiftlint version
0.26.0
$ export TOOLCHAIN_DIR=/Applications/Xcode_10_Beta_2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
$ swiftlint
sourcekit: [1:connection-event-handler:3843: 0.0000] Connection interruptsourcekit: [1:updateSemanticEditorDelay:3843: 0.0004] disabling semantic editor for 10 secondssourcekit: [1:pingService:3843: 0.0005] pinging servicesourcekitten: connection to SourceKitService restored!
Connection interrupted
Never call this for file that sourcekitd fails.: file File+Cache.swift, line 127
Abort trap: 6

```

I examined the source of swiftlint and found that the place in File+Cache that issues the fatal error. I then modified swiftlint to display the path or contents of the file with the fatal error message and ...

```
$ swiftlint
sourcekit: [1:connection-event-handler:6915: 0.0001] Connection interruptsourcekit: [1:updateSemanticEditorDelay:6915: 0.0012] disabling semantic editor for 10 secondssourcekit: [1:pingService:6915: 0.0013] pinging servicesourcekitten: connection to SourceKitService restored!
Connection interrupted
Never call this for file that sourcekitd fails. contents: #if swift(>=4.2.0)
    let version = "4.2.0"
#elseif swift(>=4.1.2)
    let version = "4.1.2"
#elseif swift(>=4.1.1)
    let version = "4.1.1"
#elseif swift(>=4.1.0)
    let version = "4.1.0"
#elseif swift(>=4.0.3)
    let version = "4.0.3"
#elseif swift(>=4.0.2)
    let version = "4.0.2"
#elseif swift(>=4.0.1)
    let version = "4.0.1"
#elseif swift(>=4.0.0)
    let version = "4.0.0"
#elseif swift(>=3.4.0)
    let version = "3.4.0"
#elseif swift(>=3.3.2)
    let version = "3.3.2"
#elseif swift(>=3.3.1)
    let version = "3.3.1"
#elseif swift(>=3.3.0)
    let version = "3.3.0"
#elseif swift(>=3.2.3)
    let version = "3.2.3"
#elseif swift(>=3.2.2)
    let version = "3.2.2"
#elseif swift(>=3.2.1)
    let version = "3.2.1"
#elseif swift(>=3.2.0)
    let version = "3.2.0"
#elseif swift(>=3.1.1)
    let version = "3.1.1"
#elseif swift(>=3.1.0)
    let version = "3.1.0"
#elseif swift(>=3.0.2)
    let version = "3.0.2"
#elseif swift(>=3.0.1)
    let version = "3.0.1"
#elseif swift(>=3.0.0)
    let version = "3.0.0"
#endif: file File+Cache.swift, line 131
Abort trap: 6
```

**Comment by marcelofabri** [comment]
This should be fixed in master.

**Comment by SlaunchaMan** [comment]
@marcelofabri I’m using SwiftLint built off of master and I’m still getting this on our Jenkins server. If it helps here’s the output:

```
[13:44:03]: ▸ sourcekit: [1:connection-event-handler:5379: 0.0000] Connection interruptsourcekit: [1:updateSemanticEditorDelay:5379: 0.0005] disabling semantic editor for 10 secondssourcekit: [1:pingService:5379: 0.0006] pinging servicesourcekitten: connection to SourceKitService restored!
[13:44:03]: ▸ Connection interrupted
[13:44:03]: ▸ Never call this for file that sourcekitd fails.: file File+Cache.swift, line 127
```

**Comment by ml-milan-vit** [comment]
It seems that in Xcode 10 Beta 3, the issue got worse. Previously, I was getting this error roughly 50% of the time, now it’s closer to 90%.

**Comment by Cellane** [comment]
It seems that in Xcode 10 Beta 3, the issue got worse. Previously, I was getting this error roughly 50% of the time, now it’s closer to 90%.

SwiftLint installed via `brew install swiftlint --HEAD` (`HEAD-1c8d5a4`).

**Comment by ctreffs** [comment]
I can confirm, that this problem occurs with swiftlint `(HEAD-1c8d5a4)` 
Xcode 10.0 beta 3 (10L201y) on macOS 10.14 Beta (18A293u) with a Swift 4.2 project.


#### Xcode Build log
```objc
...
Correcting 'AppDelegate.swift' (7/7)
Done correcting 7 files!
sourcekit: [1:connection-event-handler:10243: 0.0000] Connection interruptsourcekit: [1:updateSemanticEditorDelay:10243: 0.0009] disabling semantic editor for 10 secondssourcekit: [1:pingService:10243: 0.0010] pinging servicesourcekitten: connection to SourceKitService restored!
Connection interrupted
Never call this for file that sourcekitd fails.: file File+Cache.swift, line 127
... : 55065 Abort trap: 6           swiftlint
Command PhaseScriptExecution failed with a nonzero exit code
```
#### SwiftLint crash log
```objc
...
Thread 1 Crashed:: Dispatch queue: com.apple.root.default-qos
0   libsystem_kernel.dylib        	0x00007fff54d97eb6 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff54e35088 pthread_kill + 285
2   libsystem_c.dylib             	0x00007fff54ce4189 abort + 127
3   swiftlint                     	0x0000000105bf3e37 queuedFatalError(_:file:line:) + 375
4   swiftlint                     	0x0000000105bd4114 File.structure.getter + 308
5   swiftlint                     	0x0000000105c2efec closure #1 in variable initialization expression of static SwiftVersion.current + 796
6   swiftlint                     	0x0000000105c2ecb9 globalinit_09_8028A2A0b7E0BFCD3A13F2AE99640A32C_func33 + 9
7   libdispatch.dylib             	0x00007fff54bed4d2 _dispatch_client_callout + 8
8   libdispatch.dylib             	0x00007fff54bed48b dispatch_once_f + 41
9   swiftlint                     	0x0000000106307dd9 swift_once + 25
10  swiftlint                     	0x0000000105ce39e4 specialized static ImplicitGetterRule.nonTriggeringExamples.getter + 484
11  swiftlint                     	0x0000000105ce1333 globalinit_33_04076D872A6C6414CF30C9BF70CFAC56_func143 + 19
12  libdispatch.dylib             	0x00007fff54bed4d2 _dispatch_client_callout + 8
13  libdispatch.dylib             	0x00007fff54bed48b dispatch_once_f + 41
14  swiftlint                     	0x0000000106307dd9 swift_once + 25
15  swiftlint                     	0x0000000105ce2a31 protocol witness for static Rule.description.getter in conformance ImplicitGetterRule + 49
16  swiftlint                     	0x0000000105c2d088 specialized RuleList.init(rules:) + 168
17  swiftlint                     	0x0000000105c20401 globalinit_33_3FB051535C35994A453E379DF5C1C255_func27 + 3969
18  libdispatch.dylib             	0x00007fff54bed4d2 _dispatch_client_callout + 8
19  libdispatch.dylib             	0x00007fff54bed48b dispatch_once_f + 41
20  swiftlint                     	0x0000000106307dd9 swift_once + 25
21  swiftlint                     	0x0000000105c03cda specialized Configuration.init(path:rootPath:optional:quiet:enableAllRules:cachePath:) + 970
22  swiftlint                     	0x0000000105bfcd60 Configuration.init(path:rootPath:optional:quiet:enableAllRules:cachePath:) + 48
23  swiftlint                     	0x0000000105e68814 specialized Configuration.init(options:) + 260
24  swiftlint                     	0x0000000105e58dd6 specialized LintCommand.run(_:) + 326
25  swiftlint                     	0x0000000105e56331 protocol witness for CommandProtocol.run(_:) in conformance LintCommand + 49
26  swiftlint                     	0x0000000105b163f6 closure #1 in CommandWrapper.init<A>(_:) + 630
27  swiftlint                     	0x0000000105b190e9 partial apply for closure #1 in CommandWrapper.init<A>(_:) + 25
28  swiftlint                     	0x0000000105b16ec8 CommandRegistry.run(command:arguments:) + 248
29  swiftlint                     	0x0000000105b186a8 specialized CommandRegistry.main(arguments:defaultVerb:errorHandler:) + 520
30  swiftlint                     	0x0000000105b17090 CommandRegistry.main(defaultVerb:errorHandler:) + 48
31  swiftlint                     	0x0000000105e507ad closure #1 in  + 717
32  swiftlint                     	0x0000000105bf4210 thunk for @escaping @callee_guaranteed () -> () + 3
... 
```

#### SourceKitService crash log
```objc
...
Thread 1 Crashed:: Dispatch queue: com.apple.root.default-qos
0   com.apple.SourceKitService.1000.0.25.1_osx	0x0000000107d89630 swift::ide::SyntaxModelContext::SyntaxModelContext(swift::SourceFile&) + 1120
1   com.apple.SourceKitService.1000.0.25.1_osx	0x0000000107bda3c9 SourceKit::SwiftEditorDocument::readSyntaxInfo(SourceKit::EditorConsumer&) + 553
2   com.apple.SourceKitService.1000.0.25.1_osx	0x0000000107beac31 SourceKit::SwiftLangSupport::editorOpen(llvm::StringRef, llvm::MemoryBuffer*, bool, SourceKit::EditorConsumer&, llvm::ArrayRef<char const*>) + 2177
3   com.apple.SourceKitService.1000.0.25.1_osx	0x0000000107c1a29a handleRequestImpl(void*, std::__1::function<void (void*)>) + 11498
4   com.apple.SourceKitService.1000.0.25.1_osx	0x0000000107b993dc invocation function for block in sourcekitdServer_peer_event_handler(_xpc_connection_s*, void*) + 748
...
```

**Comment by tfonfara** [comment]
Same problem here with Xcode 10b3 (10L201y) and swiftlint 0.26.0 installed via cocoapods (checksum f6b83e8d95ee1e91e11932d843af4fdcbf3fc764).

**Comment by abbasmousavi** [comment]
Same problem here with XCode 10 beta 3 and SwiftLint installed with Cocoapods 0.26.0

**Comment by marcelofabri** [comment]
I suggest filling bugs with Apple via radar or bugs.swift.org as it looks like it's something on their side (SourceKitService crashing)

**Comment by marcelofabri** [comment]
Thanks for checking it @billThePill 🚀

**Comment by ctreffs** [comment]
I can confirm as well that this fixed the issue (swiftlint/HEAD-10a5826).
Thanks for the quick fix and the awesome tool 👍

**Comment by RyuX51** [comment]
When will the updated version hit brew?

**Comment by fssilva** [comment]
It seems this issue is back ? 

I'm getting  `Never call this for file that sourcekitd fails.: file File+Cache.swift, line 127` on `xCode 10b4(10L213o)`.

**Comment by norio-nomura** [comment]
> It seems this issue is back ?

It is affected by another SourceKit issue. [SR-8315](https://bugs.swift.org/browse/SR-8315)

**Comment by zenghao0708** [comment]
> It would be affected by another SourceKit issue. SR-8315

is there any solution for this issue? i use Xcode 10b4(10L213o) and SwiftLint 0.26.0

**Comment by norio-nomura** [comment]
[The fix](https://github.com/apple/swift/pull/18166) has already been merged to `swift-4.2-branch`.
However, new `swift-4.2-DEVELOPMENT-SNAPSHOT-*` including that fix has not come out yet.
We need to wait for the next `swift-4.2-DEVELOPMENT-SNAPSHOT-*`.

**Comment by ornithocoder** [comment]
@norio-nomura is it the same problem I faced [here](https://github.com/realm/SwiftLint/pull/2285#issuecomment-403766175)?

**Comment by norio-nomura** [comment]
@ornithocoder As far as I tested, Swift 4.1.2 is not affected by [SR-8315](https://bugs.swift.org/browse/SR-8315).
It would be good to use 
```
SOURCEKIT_LOGGING=3 swift test --filter RulesTests.testAnyObjectProtocol
```
to check what kind of request is crashing SourceKit and report it to https://bugs.swift.org.
